rcl_interfaces Log message level field does not match type of constants

The rcl_interfaces Log message level field does not match the type of constants. The constants are of type byte and the field is of type uint8.

This means the following Python code fails with an exception:
```
from rcl_interfaces.msg import Log
msg = Log()
msg.level = Log.INFO
```

To get this to work, something like this needs to be done which is a bit ridiculous:
```
from rcl_interfaces.msg import Log
msg = Log()
msg.level = int.from_bytes(Log.INFO, byteorder='little')
```

Either the level field needs to be changed to a byte type or the constants should be changed to uint8. I don't think rclpy should be expected to do a silent conversion between types to make this work correctly.

> It didn't occur to me that it would be fixed in a future rclpy version.

So this is an interesting case.  It isn't actually "fixed", though we no longer see it due to some changes in more recent code.

Sometime back in March or April, we merged a change where we disabled-by-default the run-time type checking in ROS Python message code.  We ended up doing that because that type checking was pretty expensive and actually showed up in flame graphs.

But you can see that the original problem pointed out here is still a problem in Rolling and Iron by doing the following:

```
$ export ROS_PYTHON_CHECK_FIELDS=1
$ python3
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from rcl_interfaces.msg import Log
>>> msg = Log()
>>> msg.level = Log.INFO
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/new_ros2_ws/install/rcl_interfaces/lib/python3.10/site-packages/rcl_interfaces/msg/_log.py", line 236, in level
    assert \
AssertionError: The 'level' field must be of type 'int'
```

In this case, I agree that we should make the Level constants the same type as the `level` field, which will fix this issue for good.  This isn't something we'd be able to backport, but at least it would be fixed for future versions.

--- a/rcl_interfaces/msg/Log.msg
+++ b/rcl_interfaces/msg/Log.msg
@@ -12,21 +12,21 @@
 ##
 
 # Debug is for pedantic information, which is useful when debugging issues.
-byte DEBUG=10
+uint8 DEBUG=10
 
 # Info is the standard informational level and is used to report expected
 # information.
-byte INFO=20
+uint8 INFO=20
 
 # Warning is for information that may potentially cause issues or possibly unexpected
 # behavior.
-byte WARN=30
+uint8 WARN=30
 
 # Error is for information that this node cannot resolve.
-byte ERROR=40
+uint8 ERROR=40
 
 # Information about a impending node shutdown.
-byte FATAL=50
+uint8 FATAL=50
 
 ##
 ## Fields

